File: timeliner_diagnostics.rb
if Debug
  def quit(_) $stdout.puts "#$0 ABORT: #{_}\n\n"; exit 1; end
  def warn(_) $stdout.puts "#$0 WARN: #{_}\n" end
  def info(_) $stdout.puts "#$0 info: #{_}" end
else
  # Get user's attention: report only fatality.
  # They'll ignore more than one diagnostic.
  def quit(_) $stdout.puts "#$0 ABORT: #{_}"; exit 1; end
  def warn(_) quit(_) end
  def info(_) end
end

def requireGem r, gem=r
  require r
  rescue Exception  
    quit "First, sudo gem install #{gem}"
end

# If a ruby script was retrieved as foo.txt from a webserver
# (so the webserver wouldn't mistakenly run it as a cgi script),
# remind the user to rename it from foo_rb.txt to foo.rb.
def forceFilename n
  quit "rename #$0 to #{n}" if File.basename($0) != n
end

File: timeliner_diagnostics.cpp
#include <iostream>
#include <cstdlib>
#include "timeliner_diagnostics.h"

std::string appname = "app";

#define VERBOSE
#ifdef VERBOSE
void quit(const std::string& _) { std::cout << appname + " ABORT: " + _ +"\n\n"; exit(1); }
void warn(const std::string& _) { std::cout << appname + " WARN: " + _ +"\n"; }
void info(const std::string& _) { std::cout << appname + " info: " + _ +"\n"; }
#else
// Get user's attention: report only fatality.
// They'll ignore more than one diagnostic.
void quit(const std::string& _) { std::cout << appname + " ABORT: " + _ + "\n\n"; exit(1); }
void warn(const std::string& _) { quit(_); }
void info(const std::string& _) {}
#endif
